htsget-search/src/htsget/bam_search.rs

```diff
@@ -40,6 +40,10 @@ impl VirtualPositionExt for VirtualPosition {
   /// If for some reason we can't read correctly the records we fall back
   /// to adding the maximum BGZF block size.
   fn bytes_range_end(&self, reader: &mut bam::Reader<File>) -> u64 {
+    if self.uncompressed() == 0 {
```

Comment:
Makes sense... could the user/client supply a negative range?

Reply:
No, they can't `start` and `end` in the `Query` struct are `usize`. Also in this snippet self is a `VirtualPosition` which also can't be negative.